testing/platform/fpga_hw.cpp

```diff
@@ -205,7 +205,9 @@ std::vector<std::string> test_platform::platforms(
   keys.erase(
     std::remove_if(keys.begin(), keys.end(), [drv](const std::string &n) {
       auto db = fpga_db::instance();
-      return !db->exists(n);
+      return !db->exists(n) || (drv != fpga_driver::linux_any
+                                && drv != db->get(n).driver)
+                            || db->get(n).devices.empty();
```

Comment:
devices can be empty?

Reply:
Devices *shouldn't* be empty and the db should only be populated with the detected platform when running on hardware. However, due to a bug whose source has yet to be identified, when we run on a system that has only the dfl platform, skx-p and dcp-rc also are in the db although they have no devices. This change is a workaround for now so that we can continue verifying the rest of the tests on hardware. This should only affect tests that are specific to certain platforms and only when running on hardware and with this workaround, even that case will work.